float32 x
float32 y
---
float32 err_x
float32 err_y